pandas/_libs/tslibs/parsing.pyx

```diff
@@ -0,0 +1,682 @@
+# -*- coding: utf-8 -*-
+# cython: profile=False
+# cython: linetrace=False
+# distutils: define_macros=CYTHON_TRACE=0
+# distutils: define_macros=CYTHON_TRACE_NOGIL=0
+"""
+Parsing functions for datetime and datetime-like strings.
+"""
+import sys
+import re
+
+from cpython cimport PyString_Check, PyUnicode_Check
+
+from libc.stdlib cimport free
+
+cimport cython
+from cython cimport Py_ssize_t
+
+
+from datetime import datetime
+import time
+
+import numpy as np
+cimport numpy as np
+from numpy cimport int64_t, ndarray
+np.import_array()
+
+# Avoid import from outside _libs
+if sys.version_info.major == 2:
+    string_types = basestring
+    from StringIO import StringIO
+else:
+    string_types = str
+    from io import StringIO
+
+
+# dateutil compat
+from dateutil.tz import (tzoffset,
+                         tzlocal as _dateutil_tzlocal,
+                         tzfile as _dateutil_tzfile,
+                         tzutc as _dateutil_tzutc,
+                         tzstr as _dateutil_tzstr)
+from dateutil.relativedelta import relativedelta
+from dateutil.parser import DEFAULTPARSER
+from dateutil.parser import parse as du_parse
+
+
+class DateParseError(ValueError):
+    pass
+
+_nat_strings = set(['NaT', 'nat', 'NAT', 'nan', 'NaN', 'NAN'])
```

Comment:
_nat_strings should be in datetime.pxd (or maybe in util.pxd)

Reply:
Given the recent build-based threads, I'd like to (at least for now) avoid introducing this dependency, especially for something this small.  Eventually you're right we'll want to put these constants all in one place.